src/api/spec/controllers/webui/session_controller_spec.rb

```diff
@@ -54,4 +55,71 @@
       it_behaves_like 'login'
     end
   end
+
+  context 'in kerberos mode' do
+    before do
+      stub_const('CONFIG', CONFIG.merge('kerberos_service_principal' => 'HTTP/obs.test.com@test_realm.com',
+                                        'kerberos_realm'             => 'test_realm.com',
+                                        'kerberos_mode'              => true))
+    end
+
+    context 'for a request that requires authentication' do
+      render_views
+      before do
+        get :new
+      end
+
+      context "and 'Negotiate' header is not set" do
+        it 'informs the client tool (browser) that kerberos authentication is required' do
+          expect(response.headers['WWW-Authenticate']).to eq('Negotiate')
+          expect(response.status).to eq(401)
+        end
+
+        it 'informs users about failed kerberos authentication and possible cause' do
+          expect(response.body).to have_text('Kerberos authentication required')
+          expect(response.body).to have_text('You are seeing this page, because you are ' \
```

Comment:
Is this expect really required? I would say that with the first one is enough...